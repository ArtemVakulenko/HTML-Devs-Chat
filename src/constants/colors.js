export const colors = {
  dark: {
    main: '#0E1621',
    secondary: '#17212B',
    input: '#242F3D',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#fff',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#0E1621',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#488ccf',
  },
  light: {
    main: '#FFFFFF',
    secondary: '#ECECEC',
    input: '#D1D1D1',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#000',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#FFFFFF',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#2B5278',
  },
};
