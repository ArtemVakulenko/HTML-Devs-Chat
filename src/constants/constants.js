export const langs = {
  arabian: 'ar',
  russian: 'ru',
  english: 'en',
};

export const cookieAge = 20000;
