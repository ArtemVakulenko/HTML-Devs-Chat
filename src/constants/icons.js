export const icons = {
    changeThemeIcon: '../assets/img/changeThemeIcon.png',
    defaultUserIcon: '../assets/img/defaultUserIcon.png',
    dropDownIcon: '../assets/img/dropDownIcon.png',
    logoIcon: '../assets/img/logoIcon.png',
    logOutIcon: '../assets/img/logOutIcon.png',
    settingIcon: '../assets/img/settingsIcon.png',
    closeModalDark: '../assets/img/closeModalDark.png',
    sendMessage: '../assets/img/sendMessage.png',
    showPass: '../assets/img/view.png',
    hidePass: '../assets/img/hide.png',
    themeDark: '../assets/img/moon.png',
    themeLight: '../assets/img/sun.png',
};
