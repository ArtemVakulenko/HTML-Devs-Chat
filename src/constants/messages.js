export const MESSAGES_KEYS = {
  invalidLogin: 'invalidLogin',
  invalidPassword: 'invalidPassword',
  invalidConfirm: 'invalidConfirm',
  userIsReg: 'userIsReg',
  notExist: 'notExist',
  wrongPass: 'wrongPass',
  abstract: 'something went wrong',
  successReg: 'successReg',
  successAuth: 'successAuth',
  unauth: 'unAuth',
};
