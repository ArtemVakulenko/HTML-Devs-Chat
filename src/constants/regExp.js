export const LOGIN = new RegExp(/^[a-z0-9]{3,18}/, 'i');
export const PASSWORD = new RegExp(/^[a-z0-9!@$^."â„–;%:?*\(\)-_=+]{5,20}$/, 'i');
