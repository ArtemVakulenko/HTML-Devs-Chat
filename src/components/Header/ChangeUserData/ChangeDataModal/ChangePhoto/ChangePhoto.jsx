import React from 'react';
import PropTypes from 'prop-types';
import { icons } from 'constants';
import FileIpt from 'src/components/_common/FileIpt';
import { StChangePhoto } from './styled';

const ChangePhoto = ({ img, value, setUserImg }) => (
  <StChangePhoto>
    <img src={img || icons.defaultUserIcon} />
    <FileIpt value={value} onChange={setUserImg} />
  </StChangePhoto>
);

export default ChangePhoto;

ChangePhoto.propTypes = {
  img: PropTypes.string,
  value: PropTypes.string,
  setUserImg: PropTypes.func.isRequired,
};
