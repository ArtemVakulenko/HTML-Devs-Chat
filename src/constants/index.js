import { colors } from './colors';
import { icons } from './icons';
import { MESSAGES_KEYS } from './messages';
import { url } from './url';
import { LOGIN, PASSWORD } from './regExp';
import { langs, cookieAge } from './constants';

export { url, colors, icons, MESSAGES_KEYS, LOGIN, PASSWORD, langs, cookieAge };
