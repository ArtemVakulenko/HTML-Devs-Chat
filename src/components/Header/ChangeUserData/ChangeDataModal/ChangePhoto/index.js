import { connect } from 'react-redux';
import { userImgValue } from 'store/modals/selectors';
import { setUserImg } from 'store/modals/actions';
import ChangePhoto from './ChangePhoto';

const mapStateToProps = (state) => ({
  value: userImgValue(state, 'name'),
  img: userImgValue(state, 'code'),
});
const mapDispatchToProps = {
  setUserImg,
};
export default connect(mapStateToProps, mapDispatchToProps)(ChangePhoto);
