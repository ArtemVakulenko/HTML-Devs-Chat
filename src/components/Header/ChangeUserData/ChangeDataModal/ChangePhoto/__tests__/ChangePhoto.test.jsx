import { shallow, mount } from 'enzyme';
import React from 'react';
import ChangePhoto from '../ChangePhoto';
import { icons } from '../../../../../../constants';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        button: '',
      },
    },
    theme: 'dark',
    changeTheme: jest.fn(),
  })),
}));

describe('ChangePhoto', () => {
  const props = {
    img: 'img', value: 'value', setUserImg: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = shallow(<ChangePhoto {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StChangePhoto', () => {
    const component = mount(<ChangePhoto {...props} />);
    expect(component.find('styled__StChangePhoto')).toHaveLength(1);
  });
  it('Should have img', () => {
    const component = mount(<ChangePhoto {...props} />);
    expect(component.find('img')).toHaveLength(1);
  });
  it('Should have FileIpt', () => {
    const component = mount(<ChangePhoto {...props} />);
    expect(component.find('FileIpt')).toHaveLength(1);
  });
  it('Should have src img', () => {
    const component = mount(<ChangePhoto {...props} />);
    expect(component.find('img').props().src).toBe(props.img);
  });
  it('Should have src default', () => {
    props.img = '';
    const component = mount(<ChangePhoto {...props} />);
    expect(component.find('img').props().src).toBe(icons.defaultUserIcon);
  });
});
