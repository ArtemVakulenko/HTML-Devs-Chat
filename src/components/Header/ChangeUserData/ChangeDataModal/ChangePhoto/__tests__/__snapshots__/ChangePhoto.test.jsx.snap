// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePhoto Should match snapshot 1`] = `"<div class=\\"styled__StChangePhoto-hqHpke kvzQnW\\"><img src=\\"img\\"/><label for=\\"upload\\" class=\\"styled__StFileIpt-fLkDRX iPkIEE\\">value</label><input type=\\"file\\" id=\\"upload\\" hidden=\\"\\"/></div>"`;
